projects/m5/001-budget-app/js/index.js

```diff
@@ -0,0 +1,225 @@
+//m5-001-budget-app
+
+// Class that create a budjet category with the desired name
+class Category {
+    constructor(name) {
+        //arguments
+        this.name = name;
+        this.ledger = [];
+    }
+
+    //methods
+    /**
+     * deposit allows to add an obj to the ledger with the form of {'amount': quantity, 'description':description} 
+     * @param {number} quantity the max digit of the number is 7 
+     * @param {string} description it can also be omitted
+     * @returns add an obj to the ledger list
+     */
+    deposit(quantity, description){
+        if (quantity > 99999.99){
+                throw new Error(`Deposit amount can not contain more than 7 digits.
+                The following deposit has blocked the process: amount = ${quantity}, descritpion = ${description}`)
+        }else{
+            description = description || ''; // same as: if(description===undefined){description=''}
+            return this.ledger.push({'amount': quantity, 'description':description})
+        }
+    };
+    /**
+     * same concept as deposit but the obj is added only if the balance as enough quantity
+     * @param {number} quantity the max digit of the number is 7 and is added as a negative n. to ledger
+     * @param {string} description it can also be omitted
+     * @returns {boolean} true o false
+     */
+    withdraw(quantity, description) {
+        if (quantity > 99999.99){
+            console.error(`Withdraw amount can not contain more than 7 digits`)
+            console.error(`The following withdraw has not been proceed: amount = ${quantity}, descritpion = ${description}`)
+            return false
+        }else{
+            description = description || '';
+            quantity = -quantity
+            if (this.getBalance() >= -quantity){
+                this.ledger.push({'amount': quantity, 'description':description})
```

Comment:
Why all this round of positive and negative numbers with the same name?
Check if quantity (positive) > this.getBalance()....if yes -> return false else amount:-quantity
what do you think about it?

Reply:
totally agree! usless added code.